scripts/bootloader/provision.py

```diff
@@ -32,10 +32,9 @@ def generate_provision_hex_file(s0_address, s1_address, hashes, provision_addres
     provision_data += struct.pack('H', 1) # Type "counter collection"
     provision_data += struct.pack('H', num_counters)
 
+    assert num_counter_slots_version % 2 == 0, "--num-counters-slots-version must be an even number"
```

Comment:
Would this result in a failed build so that users are not able to run menuconfig to change the option to an accepted value?

Reply:
Good question.

This assert does not prevent menuconfig from running luckily.

I tested locally as I was unsure.

I think you have to cause CMake to crash to prevent menuconfig from running.

So an assertion from CMake would be unfortunate.

But this python script is executed by ninja, not CMake.

```

    assert num_counter_slots_version % 2 == 0, "--num-counters-slots-version must be an even number"
AssertionError: --num-counters-slots-version must be an even number
[173/190] Generating zephyr/b0_container.hex
FAILED: zephyr/provision.hex /home/sebo/ncs/nrf/tests/subsys/bootloader/boot_chains/build/zephyr/provision.hex 
cd /home/sebo/ncs/nrf/tests/subsys/bootloader/boot_chains/build/zephyr && /usr/bin/python3.10 /home/sebo/ncs/nrf/scripts/bootloader/provision.py --s0-addr 0x8000 --s1-addr 0x84000 --provision-addr 0xff8100 --public-key-files /home/sebo/ncs/nrf/tests/subsys/bootloader/boot_chains/build/zephyr/nrf/subsys/bootloader/generated/public.pem --output /home/sebo/ncs/nrf/tests/subsys/bootloader/boot_chains/build/zephyr/provision.hex --num-counter-slots-version 3 --max-size 0x280
[175/190] Generating dev_handles.c
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/sebo/ncs/nrf/tests/subsys/bootloader/boot_chains/build
sebo@sebo:~/ncs/nrf/tests/subsys/bootloader/boot_chains$ cd build/
cd build/
sebo@sebo:~/ncs/nrf/tests/subsys/bootloader/boot_chains/build$ xterm -e ninja menuconfig
xterm -e ninja menuconfig
sebo@sebo:~/ncs/nrf/tests/subsys/bootloader/boot_chains/build$ ninja
ninja
[0/1] Re-running CMake...

```